### PR TITLE
feat: use figma light theme variables

### PR DESCRIPTION
**Pull request by garrettbear**
## **Description**
Update CSS-in-JS to use Figma Variables brand colors and add on brand evolution. Use figma variable light and dark theme json that's tied to the new brand colors. 

- Added "some" automation, there is a script that can be used for the brand colors and themes that will output about 90% of what is needed but does require some manual changes (e.g. when a color is 400-10%)

## **Related issues**

Fixes: #660 #661 

## **Manual testing steps**

1. Go to this page...
2.
3.

## **Screenshots/Recordings**

https://github.com/MetaMask/design-tokens/assets/26469696/7b83716f-66ac-4224-ba9b-2836502535a5

## **Pre-merge author checklist**

- [ ] I’ve followed [MetaMask Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've clearly explained what problem this PR is solving and how it is solved.
- [ ] I've linked related issues
- [ ] I've included manual testing steps
- [ ] I've included screenshots/recordings if applicable
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.
- [ ] I’ve properly set the pull request status:
  - [ ] In case it's not yet "ready for review", I've set it to "draft".
  - [ ] In case it's "ready for review", I've changed it from "draft" to "non-draft".

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: docs/BrandColorsBrandEvolution.stories.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ type Story = StoryObj<typeof ColorSwatchGroup>;
 
 export const Figma: Story = {
   render: () => {
-    const { brandColor } = useJsonColor();
+    const { brandColor } = useJsonColor(true);
```
Comment (garrettbear) [inline docs/BrandColorsBrandEvolution.stories.tsx:24]: Passing true will enable the useJsonColor function to use brandEvolution

File: docs/ThemeColors.stories.tsx [hunk 2]
```diff
@@ -40,12 +38,7 @@ export const FigmaDarkTheme = {
           padding: '1rem',
         }}
       >
-        <ColorSwatchGroup
-          swatchData={darkTheme}
-          borderColor={darkTheme.border.muted.value}
-          textBackgroundColor={darkTheme.background.default.value}
-          textColor={darkTheme.text.default.value}
-        />
+        <ColorSwatchGroup swatchData={darkTheme} theme="#24272a" />
```
Comment (garrettbear) [inline docs/ThemeColors.stories.tsx:41]: theme="#24272a" sets the background for dark mode color

File: docs/components/ColorSwatch/ColorSwatch.tsx [hunk 3]
```diff
@@ -37,7 +37,7 @@ export const ColorSwatch: FunctionComponent<ColorSwatchProps> = ({
       style={{
         height: 120,
         backgroundColor: color,
-        border: `2px solid ${borderColor}`,
+        border: `1px solid ${borderColor}`,
```
Comment (garrettbear) [inline docs/components/ColorSwatch/ColorSwatch.tsx:40]: just a little style improvement

File: docs/utils/useJsonColor.ts [hunk 4]
```diff
@@ -23,24 +24,25 @@ type CompiledColors = {
...
+
 /**
  * Custom hook for compiling color themes from Figma JSON definitions.
  * Merges brand, light, and dark theme color settings into a single object.
  *
+ * @param useEvolutionColors - Prop to use the brand evolution colors instead of the default brand colors.
  * @returns Object containing compiled color themes.
  */
-export const useJsonColor = (): CompiledColors => {
+export const useJsonColor = (useEvolutionColors = false): CompiledColors => {
```
Comment (garrettbear) [inline docs/utils/useJsonColor.ts:37]: when passing true it will then use the brandEvolution colors

File: src/js/brandColor/brandColor.test.ts [hunk 5]
```diff
@@ -1,7 +1,7 @@
 /* eslint-disable @typescript-eslint/no-var-requires, @typescript-eslint/no-require-imports */
 import { brandColor } from './brandColor';
 
-const designTokens = require('../../figma/brandColorBrandEvolution.json');
```
Comment (garrettbear) [inline src/js/brandColor/brandColor.test.ts:4]: fixed little mistake from when trying out the brandEvolution

File: docs/BrandColors.stories.tsx [hunk 6]
```diff
@@ -47,18 +46,8 @@ export const CSS: Story = {
...
-      {/* Mapping through each brand color and rendering a ColorSwatch component for it */}
-      {Object.entries(brandColorJS).map(([name, color]) => (
-        <ColorSwatch key={name} color={color} name={`brandColor.${name}`} />
-      ))}
-    </div>
-  ),
+  render: () => {
+    const { brandColor } = useJsonColor();
+    return <ColorSwatchGroup swatchData={brandColor} />;
+  },
```
Comment (georgewrmarshall) [inline docs/BrandColors.stories.tsx:52]: The idea here is that we are displaying the JavaScript values so we can do a visual check of the JS tokens what is the reason here to change it to use the JSON?

Screenshot 2024-05-17 at 2 25 13 PM
Screenshot 2024-05-17 at 2 24 56 PM

File: docs/BrandColorsBrandEvolution.stories.tsx [hunk 7]
```diff
@@ -47,18 +46,8 @@ export const CSS: Story = {
...
-      {/* Mapping through each brand color and rendering a ColorSwatch component for it */}
-      {Object.entries(brandColorJS).map(([name, color]) => (
-        <ColorSwatch key={name} color={color} name={`brandColor.${name}`} />
-      ))}
-    </div>
-  ),
+  render: () => {
+    const { brandColor } = useJsonColor(true);
+    return <ColorSwatchGroup swatchData={brandColor} />;
+  },
```
Comment (georgewrmarshall) [inline docs/BrandColorsBrandEvolution.stories.tsx:52]: Same here?

File: docs/components/ColorSwatchGroup/ColorSwatchGroup.tsx [hunk 8]
```diff
@@ -13,82 +13,157 @@ interface ColorSwatchGroupProps {
...
+    return g[1].toUpperCase();
+  });
 }
 
 export const ColorSwatchGroup: React.FC<ColorSwatchGroupProps> = ({
   swatchData,
   borderColor,
   textBackgroundColor = 'transparent',
-  textColor = '#212529',
+  theme = '#ffffff',
```
Comment (georgewrmarshall) [inline docs/components/ColorSwatchGroup/ColorSwatchGroup.tsx:36]: Looks like `textColor` has been removed here but its still in the `ColorSwatchGroupProps` enum

File: src/js/themesBrandEvolution/darkTheme/colors.ts [hunk 9]
```diff
@@ -0,0 +1,77 @@
+import { brandColor } from '../../brandColorBrandEvolution';
+import type { ThemeColors } from '../types';
+
+export const colors: ThemeColors = {
+  background: {
+    default: brandColor.grey800, // For default neutral backgrounds
```
Comment (georgewrmarshall) [inline src/js/themesBrandEvolution/darkTheme/colors.ts:6]: We many not need these comments if they are present in the types. I think it's the types that enables them to display on hover in Visual Studio

**Review by georgewrmarshall — COMMENTED** [review]
Excellent work on automations and integrating the brand evolution colors into the JS tokens 🎉 However, it seems there are some issues with the JS tokens displaying in Storybook. It looks like they've been updated to display the Figma JSON? The purpose of having Figma, CSS, and JS stories is to provide a visual representation of all formats, allowing us to check for visual regressions. Once we update that we can merge so we can start testing

**Review by georgewrmarshall — APPROVED** [review]

**Pull request merged**